### PR TITLE
added hardcoded gemm interface

### PR DESCRIPTION
**Pull request by theoden8**
As requested by @mehdi-goli and @lukeiwanski, I created an interface to multiply matrices of hardcoded size.

### CODE REVIEW

File: include/operations/blas3_trees_gemm.hpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*************************************************************************** 
+/***************************************************************************
  *  @license
  *  Copyright (C) 2017 Codeplay Software Limited
  *  Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (theoden8) [inline include/operations/blas3_trees_gemm.hpp]: This file was not intended to be changed. I will consider resubmitting the pull request.

**Comment by gflegar** [comment]
Just realized: you never actually test if this works or not. Right?

File: tests/blas3_gemm_test.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  Executor<SYCL> ex(q);
+
+  {
+    buffer<T, 1> mba(ma, range<1>{m * k});
+    buffer<T, 1> mbb(mb, range<1>{k * n});
+    buffer<T, 1> mbc(mc, range<1>{m * n});
+    char _tra = 'N', _trb = 'N';
+    T alpha = .54, beta = .39;
+    std::cout << "running kernel" << std::endl;
+    _gemm(ex, _tra, _trb, m, k, n, alpha, mba, m, mbb, n, beta, mbc, k);
```
Comment (gflegar) [inline tests/blas3_gemm_test.cpp:38]: Found the problem, should be `m, n, k` here.

Comment (theoden8) [reply tests/blas3_gemm_test.cpp:38]: I test that it doesn't fail, the testing was already yours there.
In this case mkn are all same, it shouldnt affect correctness.

Comment (gflegar) [reply tests/blas3_gemm_test.cpp:38]: If you don't change m, k, n it works, it stops working if you do (like I did).

Comment (theoden8) [reply tests/blas3_gemm_test.cpp:38]: My bad, sorry for that.

**Pull request merged**